Postal/IMAPSession.swift

```diff
@@ -214,26 +219,26 @@ final class IMAPSession {
     func configureIdentity() throws {
         guard capabilities.contains(.Id) else { return }
         
-        var serverId: UnsafeMutablePointer<mailimap_id_params_list> = nil
+        var serverId: UnsafeMutablePointer<mailimap_id_params_list>? = nil
         try mailimap_id(imap, nil, &serverId).toIMAPError?.check()
         defer { mailimap_id_params_list_free(serverId) }
         
-        guard let list = serverId.optional?.idpa_list else { return }
+        guard let list = serverId?.pointee.idpa_list else { return }
         
         var dic = [String:String]()
         sequence(list, of: mailimap_id_param.self)
             .flatMap { (param: mailimap_id_param) -> (String, String)? in
-                guard let key = String.fromCString(param.idpa_name) else { return nil }
-                guard let value = String.fromCString(param.idpa_value) else { return nil }
+                guard let key = String.fromUTF8CString(param.idpa_name) else { return nil }
+                guard let value = String.fromUTF8CString(param.idpa_value) else { return nil }
                 return (key, value)
             }.forEach { dic[$0] = $1 }
         
         serverIdentity = IMAPIdentity(dic)
     }
     
-    private func checkCertificateIfNeeded() throws -> Bool {
+    @discardableResult fileprivate func checkCertificateIfNeeded() throws -> Bool {
```

Comment:
indeed, the bool parameter is unused (return true or throws), we should remove it